WordPress/wporg-mu-plugins
# New Block: Local Navigation container

This introduces a new container block for the Local Navigation in the header. The purpose of this is mostly to keep the shared styles contained, while also giving us a way to handle any custom functionality (not needed right now, but it would have helped with the breadcrumbs issue WordPress/wporg-main-2022#325 — while that doesn't apply to the new design, something like that could come up again).

The proposed designs show a small local header on interior pages. This block would handle the black-background section.

![Section Header L2](https://github.com/WordPress/wporg-mu-plugins/assets/541093/86bd1e5b-483b-4a38-9284-1556ccae3921)

See https://github.com/WordPress/wporg-mu-plugins/issues/465

**Screenshots**

By default, this inserts a full-width blue banner, with breadcrumbs and a navigation block

![default](https://github.com/WordPress/wporg-mu-plugins/assets/541093/fbe3c56f-ffa7-4b86-8fe4-a33b9abdbd40)

The brush stroke style variation is also available (I'm not sure if this will carry over to the redesign, but it is currently used on the Downloads subpages).

![after-download-nightly](https://github.com/WordPress/wporg-mu-plugins/assets/541093/d21e86d7-4059-4769-8be2-0dfcc7764ff6)

And the colors can be fully customized, background, text, border, etc

![after-about-reqs](https://github.com/WordPress/wporg-mu-plugins/assets/541093/3e03df2b-4fb8-4a22-858d-a272baa98760)

The banner scrolls, sticking under the sticky-global-header.

![after-about-reqs-scroll](https://github.com/WordPress/wporg-mu-plugins/assets/541093/5b7c016c-f190-408c-b275-a7f817190464)

Small/tablet sized screens 

![mid-screen](https://github.com/WordPress/wporg-mu-plugins/assets/541093/75a9a60a-dcc5-4a7a-97bf-e5fbac85aba4)

On small screens, the menu collapses (in these screenshots I've had it use the text display so we don't get two stacked hamburgers 🍔 ). When open, it takes over the whole screen (native GB behavior). We could iterate on this some more for a less "in your face" menu, if needed.

| Small screen, closed | Small screen, open |
|---|---|
| ![small-screen](https://github.com/WordPress/wporg-mu-plugins/assets/541093/f0d01610-536f-4a3a-9fa8-b7cd07d47c8e) | ![small-screen-open](https://github.com/WordPress/wporg-mu-plugins/assets/541093/28451d60-88f9-4f8d-a6bb-c816bc6f4b76) |

Menu with dropdowns are also supported, but not likely to be used on the redesign.

![with-dropdown](https://github.com/WordPress/wporg-mu-plugins/assets/541093/9bceb588-321d-4df5-8e60-1d58e3e8a060)

**To Test**

- Try the PR for wporg-main-2022 using this block (TBD)
- Try using the site editor to add the block to a template
- Or mock up the block in the post editor, and copy it into a template file
- It should display as expected/nothing weird should happen 🙂
